src/app/guards/auth-verified.guard.ts

```diff
@@ -10,10 +12,18 @@ export const AuthVerifiedGuard: CanActivateFn = (route, state) => {
     return loggedInResult;
   }
 
-  // const router = inject(Router);
-
+  const userService = inject(UserService);
+  const router = inject(Router);
+  userService.userProfile().subscribe({
+    next: (data) => {
+      if (data.is_phone_verified === false) {
+        router.navigate(['/verify-otp']);
+        sessionStorage.setItem('redirectUrl', state.url);
```

Comment:
I don't think we need to handle the redirects. Let's keep this implementation simple